src/pipert2/core/base/pipe.py

```diff
@@ -38,6 +41,7 @@ def __init__(self, network: Network, logger: Logger, data_transmitter: DataTrans
         self.flows = {}
         self.event_board = EventBoard()
         self.default_data_transmitter = data_transmitter
+        self.wires: List[Wire] = []
```

Comment:
Maybe the wires data member should be a dict instead of list?
The key will be the source_routine and you won't have 2 wires with the same source routines.
This way, if i create a new link with the same source it will override the first one.